msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -0,0 +1,258 @@
+                        "text": "Please enter the following information:"
+                    },
+                    {
+                        "type": "TextBlock",
+                        "text": "Name"
+                    },
+                    {
+                        "type": "Input.Text",
+                        "spacing": "None",
+                        "title": "New Input.Toggle",
+                        "placeholder": "Placeholder text"
+                    },
+                    {
+                        "type": "TextBlock",
+                        "text": "Date of birth"
+                    },
+                    {
+                        "type": "Input.Date",
+                        "spacing": "None",
+                        "title": "New Input.Toggle"
+                    }
+                ],
+                "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+                "version": "1.0"
+            }
+        },
+        "type": "continue"
+    }
+}
+```
+
+The bot can also respond with an auth/config response if the user needs to authenticate or configure the extension before getting the user input.
+
+### Dynamic input using a web view
+
+In this method, your service can show an `<iframe>` based widget to show any custom UI and collect user input. For this approach, set the `fetchTask` parameter to `true` in the manifest.
+
+Just like in the adaptive card flow your service will be send a fetchTask event and needs to respond with a URL based [task module response](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/task-modules/task-modules-overview#the-taskinfo-object). Below is an sample response with an adaptive card:
+
+```json
+{
+    "task": {
+        "value": {
+            "url": "http://mywebapp.com/input"
+        },
+        "type": "continue"
+    }
+}
+```
+
+The bot can also respond with an auth/config response if the user needs to authenticate or configure the extension before getting the user input.
+
+## Responding to submit
+
+Once a user completes entering their input your bot will receive a `composeExtension/submitAction` event with the command id and parameter values set.
+
+There are three different expected responses to a submitAction.
+
+### Task Module response
+
+This is used when your extension needs to chain dialogs together to get more information. The response is exactly the same as `fetchTask` mentioned earlier.
+
+### Compose extension auth/config response
+
+This is used when your extension needs to either authenticate or configure in order to continue. See [authentication section](~/concepts/messaging-extensions/search-extensions#Authentication) in the search section for more details.
+
+### Compose extension result response
+
+This used to insert a card into the compose box as a result of a the command. It's the same response that's used in the search command, but it's limited to one command in the array.
```

Comment:
One command in the array? That doesn't make any sense. Don't you mean one card or one result in the array?